examples/IB/explicit/ex7/example.cpp

```diff
@@ -391,95 +419,81 @@ main(int argc, char* argv[])
-                global_coordinates.resize(total_nodes * NDIM); // resize the receive buffer on the root process to
-                                                               // total_nodes*NDIM
+                global_coordinates.resize(total_nodes); // resize the receive buffer on the root process to receive the
+                                                        // Lag indices and the coordinates
 
                 // Setup the arrays for the counts and displacements
                 std::vector<int> count(n_proc);
                 std::vector<int> displacements(n_proc);
                 displacements[0] = 0;
-                count[0] = NDIM * num_nodes;
+                count[0] = num_nodes;
                 int sum_elem = 0;
                 for (int j = 1; j < n_proc; j++)
                 {
-                    sum_elem = sum_elem + NDIM * nodes_per_proc[j - 1];
-                    count[j] = NDIM * nodes_per_proc[j];
+                    sum_elem = sum_elem + nodes_per_proc[j - 1];
+                    count[j] = nodes_per_proc[j];
                     displacements[j] = sum_elem;
                 }
 
                 // Gather the data
 
                 MPI_Gatherv(local_coordinates.data(),
                             local_coordinates.size(),
-                            MPI_DOUBLE,
+                            mpi_lag_coords_type,
                             global_coordinates.data(),
                             count.data(),
                             displacements.data(),
-                            MPI_DOUBLE,
+                            mpi_lag_coords_type,
                             0,
                             IBTK_MPI::getCommunicator());
             }
             else
             { // specify what's done on the non-root processes
                 MPI_Gatherv(local_coordinates.data(),
                             local_coordinates.size(),
-                            MPI_DOUBLE,
+                            mpi_lag_coords_type,
                             NULL,
                             NULL,
                             NULL,
-                            MPI_DOUBLE,
+                            mpi_lag_coords_type,
                             0,
                             IBTK_MPI::getCommunicator());
             }
             IBTK_MPI::barrier();
-            // Now if we are on the root process, let's go ahead and print out the coordinates to "Coordinates.txt"
+            // Now if we are on the root process, let's go ahead and sort the global_coordinates vector according to the
+            // lag index and print out the coordinates to "IBCoordinates.txt"
             if (rank == 0)
             {
                 fout.unsetf(ios_base::showpos);
                 fout.setf(ios_base::scientific);
                 fout.precision(5);
 
-                for (int l = 0; l < total_nodes; l++)
+                std::stable_sort(global_coordinates.begin(), global_coordinates.end());
```

Comment:
```suggestion
                std::stable_sort(global_coordinates.begin(), global_coordinates.end(),
                [](const Lag_Coords &a, const Lag_Coords &b) {return a.lag_idx < b.lag_idx;});
```
Do we need a stable sort? Could we ever have two structs with equal Lagrangian indices?